fsspec/implementations/ftp.py

```diff
@@ -79,12 +79,10 @@ def ls(self, path, detail=True):
         out = []
         if path not in self.dircache:
             try:
-                out = list(self.ftp.mlsd(path))
+                out = [(fn, details) for (fn, details) in self.ftp.mlsd(path) if fn not in ['.' '..']]
```

Comment:
The string here is "...", not what you intended; but I think the functionality is exactly the same as before, no? What about the "pdir", "cdir" that you initially wanted to guard against?

Reply:
Yes- sorry!